packages/cli/generation/ir-generator/src/converters/services/convertHttpResponse.ts

```diff
@@ -53,3 +75,95 @@ function constructStreamingResponseChunkType(
         return StreamingResponseChunkType.json(file.parseTypeReference(typeReference));
     }
 }
+
+function typeReferenceHasProperty(
```

Comment:
historically what we have done is make this a `fern check` rule and then assume its all good when we get here

Reply:
this way we can keep validation somewhat separate from `generateIntermediateRepresentation.ts` and we don't have to reimplement -- since you have the code here do you just want to convert this into a fern check rule right now?

Reply:
it looks great!